### ISSUE TITLE
Shouldn't SubscribeOnMainThread do what ObserveOnMainThread does and vice versa?

### ISSUE THREAD

**Issue by doctorpangloss** [neuecc/UniRx#355]
I feel like these things are named backwards. Intuitively, `SubscribeOnMainThread` to me means execute the callback I passed to `subscribe` on the main thread.

**Comment by MystikalPooka** [comment]
callbacks passed to subscribe are actually parts of the *observer*, not part of the subscription. Subscribe just has overloads to provide convenience methods for OnNext, OnError, and On Complete. What you are thinking of (the callbacks passed to these overloads) are not part of hooking up of the events ("subscribing") that takes part when using SubscribeOnMainThread. Subscribe means "start listening now", Observe means "do this when I receive a sequence". 

TL;DR, I think they are named perfectly well, and they match the implementation of RX that uniRX is based on.
